File: sdk/java-sdk-protobuf-testkit/src/main/java/kalix/javasdk/testkit/DeferredCallDetails.java
/*
 * Copyright (C) 2021-2024 Lightbend Inc. <https://www.lightbend.com>
 */

package kalix.javasdk.testkit;

import kalix.javasdk.Metadata;

public interface DeferredCallDetails<I, O> {
  /** @return The forwarded message */
  I getMessage();

  /** @return Any metadata attached to the call */
  Metadata getMetadata();

  /** @return The name of the service being called */
  String getServiceName();

  /** @return The method name being called */
  String getMethodName();
}

File: sdk/scala-sdk-protobuf-testkit/src/main/scala/kalix/scalasdk/testkit/DeferredCallDetails.scala
/*
 * Copyright (C) 2021-2024 Lightbend Inc. <https://www.lightbend.com>
 */

package kalix.scalasdk.testkit

import kalix.scalasdk.DeferredCall

trait DeferredCallDetails[I, O] extends DeferredCall[I, O] {

  /** @return The name of the service being called */
  def serviceName: String

  /** @return The method name being called */
  def methodName: String
}
